Add more description to hasDataFile relationship type

bact opened the pull request:
- Add description to `hasDataFile`
  - Add list of cases that an author should consider using a more specific relationship type 
  - This is to resolve #690
- Also remove double "is" from `hasAddedFile`

bact committed:
Add more description to hasDataFile

Signed-off-by: Arthit Suriyawongkul <arthit@gmail.com>

bact committed:
Revise sentences

Signed-off-by: Arthit Suriyawongkul <arthit@gmail.com>

rgopikrishnan91 commented on model/Core/Vocabularies/RelationshipType.md:
```diff
@@ -43,11 +44,11 @@ name completes the sentence:
 - hasAssociatedVulnerability: (Security) Used to associate a `from` Artifact with each `to` Vulnerability
 - hasConcludedLicense: The `from` Software Artifact is concluded by the SPDX data creator to be governed by each `to` license
-- hasDataFile: The `from` Element treats each `to` Element as a data file
+- hasDataFile: The `from` Element treats each `to` Element as a data file. A data file is a data resource that the `from` Element can use to fulfil its expected functinality or behavior. It can be a generic database fie, a log file, an AI model file, a calibration data file, a temporary file, etc. For AI training data, test data, other test artifacts, configuration data, Build input data, and Bui ...
```
1. Generic database fie --> File 
2. I am unsure if saying hasDataFile is the right name? Should it hasAsset? because hasDataFile for a model executable and log file seems odd and non-intitutive. 
3. Also, I am unsure and maybe we should clarify how its different from dependsOn relationship?

bact replied:
On (2), we cannot change the name in 3.0.x, it is already published. But it is possible in 3.1.

(3) It will be helpful to have that. One thing I can see is that `hasDataFile` doesn't imply dependency. Without the `to` Element, the `from` Element can still functional.

rgopikrishnan91 replied:
Okay. I am okay with it and maybe we can note down somewhere it should be hasArtificat as the name. But otherwise, I approve

bennetkl commented:
It might be a good idea to open 3.1 PR to make the name change (if the anser to Arit's querstion, is yes we can make a name change in that 

    On Tuesday, July 30, 2024 at 03:03:38 p.m. EDT, Gopi Krishnan Rajbahadur ***@***.***> wrote:  
 
 


@rgopikrishnan91 commented on this pull request.

In model/Core/Vocabularies/RelationshipType.md:
>  - hasAssessmentFor: (Security) Relates a `from` Vulnerability and each `to` Element(s) with a security assessment. To be used with `VulnAssessmentRelationship` types
 - hasAssociatedVulnerability: (Security) Used to associate a `from` Artifact with each `to` Vulnerability
 - hasConcludedLicense: The `from` Software Artifact is concluded by the SPDX data creator to be governed by each `to` license
-- hasDataFile: The `from` Element treats each `to` Element as a data file
+- hasDataFile: The `from` Element treats each `to` Element as a data file. A data file is a data resource that the `from` Element can use to fulfil its expected functinality or behavior. It can be a generic database fie, a log file, an AI model file, a calibration data file, a temporary file, etc. For AI training data, test data, other test artifacts, configuration data, Build input data, and Build output data, please consider using `trainedOn`, `testedOn`, `hasTest`, `configures`, `hasInputs`, and `hasOutputs` types, respectively.

Okay. I am okay with it and maybe we can note down somewhere it should be hasArtificat as the name. But otherwise, I approve

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

bact committed:
Revise hasDataFile desc

"A data file is an artifact that stores data" -- drawn from a summary of File (subclass of SoftwareArtifact): "any object that stores content"

Signed-off-by: Arthit Suriyawongkul <arthit@gmail.com>

bact committed:
Add index file

Signed-off-by: Arthit Suriyawongkul <arthit@gmail.com>

bact commented:
Typo fixed.

bact committed:
Add dependency statement

Signed-off-by: Arthit Suriyawongkul <arthit@gmail.com>
Co-Authored-By: Gopi Krishnan Rajbahadur <7269497+rgopikrishnan91@users.noreply.github.com>

bact committed:
Fix cases

Signed-off-by: Arthit Suriyawongkul <arthit@gmail.com>

bact committed:
Merge branch 'spdx:main' into update-hasDataFile-desc

bact committed:
Merge branch 'main' into update-hasDataFile-desc

Signed-off-by: Arthit Suriyawongkul <arthit@gmail.com>

gopirajbahadur commented:
LGTM, I approve

bact committed:
Merge branch 'spdx:main' into update-hasDataFile-desc
